radix-engine/src/transaction/builder.rs

```diff
@@ -486,6 +487,7 @@ impl<'a, A: AbiProvider> TransactionBuilder<'a, A> {
                 args: vec![
                     SmartValue::from(ids.clone()),
                     SmartValue::from(*resource_address),
+                    SmartValue::from(Rid(0))
```

Comment:
As we're reserving `Rid(0)` and `Bid(1)` for signature, we should update the `IDAllocator` employed by `TransactionBuilder` to start counting from `2`.

Reply:
Ah yes, was wondering about this.